honeybee_vtk/data.py

```diff
@@ -0,0 +1,161 @@
+        else:
+            raise ValueError(
+                f'Object name should be from these {AcceptedValues.names.value}.'
+                f' Instead got {v}.'
+            )
+
+    @validator('file_paths')
+    def validate_paths(cls, v: List[str]) -> List[str]:
+        if all([Path(path).is_file() for path in v]):
+            return v
+        else:
+            raise ValueError(
+                'File paths are not valid.'
+            )
+
+    def cross_check_data(self, model: Model) -> bool:
+        """Cross check data with model.
+
+        For grids, it will be checked if the number of data files and the names of the
+        data files match with the grid identifiers. For other than grid objects, it will
+        be checked that only one data file is provided and the length of the data file
+        matches the length of Polydata in the model.
+
+        Args:
+            model: A honeybee-vtk model object.
+
+        Returns:
+            A boolean value.
+        """
+
+        # if object name is "grid" check that the name of files match the grid names
+        if self.object_name == AcceptedValues.names.value[-1]:
+
+            assert len(model.sensor_grids.data) > 0, 'Sensor grids are not loaded on'\
+                ' this model. Reload them using grid options.'
+
+            grid_names = [grid.identifier for grid in model.sensor_grids.data]
+            file_names = [Path(path).name.split('.')[0] for path in self.file_paths]
+            if len(grid_names) != len(file_names) or grid_names != file_names:
+                raise ValueError(
+                    'The number of files and the file names must match the grid'
+                    ' identifiers in HBJSON.'
+                )
+            return True
+
+        # if object_name is other than grid check that length of data matches the length
+        # of data in the model for that object.
+        elif self.object_name in AcceptedValues.names.value[:-1]:
+
+            if len(self.file_paths) == 0:
+                raise ValueError(
+                    'File path not found in the config file.'
+                )
+            elif len(self.file_paths) > 1:
+                raise ValueError(
+                    'Only one file path needs to be provided in order to load data on'
+                    f' {self.object_name}. Multiple files are provided in the config'
+                    ' file.'
+                )
+
+            file = open(self.file_paths[0], "r")
+            nonempty_line_count = len(
+                [line.strip("\n") for line in file if line != "\n"]
+            )
+            file.close()
+
+            if nonempty_line_count != len(
+                    model.get_modeldataset_from_string(self.object_name).data):
+                raise ValueError(
+                    'The length of data in the file does not match the number of'
+                    f' {self.object_name} objects in the model.'
+                )
+            return True
```

Comment:
What if a file has a header? Most of Radiance files do have headers actually.

Reply:
I think this one requires a separate discussion and scoping.

Reply:
Let's assume there will be no headers for now. I think all these checks will become an issue soon and we have to remove them anyways.